controllers/cloudstackfailuredomain_controller_test.go

```diff
@@ -49,14 +48,83 @@ var _ = Describe("CloudStackFailureDomainReconciler", func() {
 					arg1.(*infrav1.CloudStackZoneSpec).Network.Type = cloud.NetworkTypeShared
 				}).MinTimes(1)
 
-			tempfd := &infrav1.CloudStackFailureDomain{}
-			Eventually(func() bool {
-				key := client.ObjectKeyFromObject(dummies.CSFailureDomain1)
-				if err := k8sClient.Get(ctx, key, tempfd); err == nil {
-					return tempfd.Status.Ready
-				}
-				return false
-			}, timeout).WithPolling(pollInterval).Should(BeTrue())
 		})
+
+		It("Should set failure domain Status.Ready to true.", func() {
+			assertFailureDomainCreated()
+		})
+		It("Should delete failure domain if no VM under this failure domain.", func() {
```

Comment:
Let's be more clear about the description of the test here. Technically the failureDomain has a deletion timestamp, and the real crutch of the test is to check that the finalizer is removed.

Maybe something like `It("Should remove failure domain finalizer when no VMs are placed in it.", ...`